### PR TITLE
[DOCUMENTATION] add ColumnLevelLineage facet doc

### PR DESCRIPTION
**Pull request by julienledem**
Adding documentation for the Column-level-lineage facet

Signed-off-by: Julien Le Dem <julien@apache.org>

### CODE REVIEW

File: spec/OpenLineage.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ Example of valid name is `BigQueryStatisticsJobFacet` and it's key `bigQuery_sta
 
 - **version**: Captures the dataset version when versioning is defined by database (ex. Iceberg snapshot ID)
 
-- **columnLineage**: Captures the column-level lineage
+- [**columnLineage**](facets/ColumnLineageDatasetFacet.md): Captures the column-level lineage
```
Comment (howardyoo) [inline spec/OpenLineage.md:136]: `**[columnLineage](facets/ColumnLineageDatasetFacet.md)** :`
might be correct.

**Review by howardyoo — CHANGES_REQUESTED** [review]

File: spec/OpenLineage.md (hunk 1 above)
Comment (julienledem) [reply spec/OpenLineage.md:136]: this looks like it works? https://github.com/OpenLineage/OpenLineage/blob/ff0d87d30ed6c9fe39472788948266a6d3190585/spec/OpenLineage.md#dataset-facets

Comment (howardyoo) [reply spec/OpenLineage.md:136]: > this looks like it works? https://github.com/OpenLineage/OpenLineage/blob/ff0d87d30ed6c9fe39472788948266a6d3190585/spec/OpenLineage.md#dataset-facets

I tried your format on number of different markdown editors, and they all seem to work!! So, I guess it's perfectly fine!

File: spec/facets/ColumnLineageDatasetFacet.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          ]
+        },
+        "columnLineage": {
+          "a": {
+            "inputFields": [
+              {namespace: "my-datasource-namespace", name: "instance.schema.table", "field": "ia"},
+              ... other inputs
+            ],
+            transformationDescription: "identical",
+            transformationType: "IDENTITY"
```
Comment (pawel-big-lebowski) [inline spec/facets/ColumnLineageDatasetFacet.md:77]: Perhaps it's worth mentioning here difference between `transformationDescription`/`transformationType` and possible `transformationType` values.

**Review by pawel-big-lebowski — APPROVED** [review]
LGTM, minor comment added.

File: spec/facets/ColumnLineageDatasetFacet.md (hunk 2 above)
Comment (julienledem) [reply spec/facets/ColumnLineageDatasetFacet.md:83]: 👍🏻

**Pull request merged**